joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/ErbTests.scala

```diff
@@ -0,0 +1,194 @@
+package io.joern.rubysrc2cpg.querying
+
+import io.joern.x2cpg.Defines
+import io.joern.x2cpg.frontendspecific.rubysrc2cpg.Constants
+import io.joern.rubysrc2cpg.testfixtures.RubyCode2CpgFixture
+import io.shiftleft.codepropertygraph.generated.{ControlStructureTypes, DispatchTypes, Operators}
+import io.shiftleft.codepropertygraph.generated.nodes.{Block, Call, FieldIdentifier, Identifier, Literal}
+import io.shiftleft.semanticcpg.language.*
+
+class ErbTests extends RubyCode2CpgFixture {
+  "Complete ERB processing" should {
+    val cpg = code(
+      """
+        |app_name: <%= ENV['APP_NAME'] %>
+        |version: <%= ENV['APP_VERSION'] %>
+        |
+        |database:
+        |  host: <%= ENV['DB_HOST'] %>
+        |  port: <%= ENV['DB_PORT'] %>
+        |
+        |<% if ENV['USE_REDIS'] == 'true' %>
+        |redis:
+        |  host: <%= ENV['REDIS_HOST'] %>
+        |  port: <%= ENV['REDIS_PORT'] %>
+        |<% end %>
+        |""".stripMargin,
+      "test.erb"
+    )
+
+    "Contain a call to <operator>.conditional" in {
+      inside(cpg.call.name(Operators.conditional).l) {
+        case _ :: Nil =>
+        // Do nothing - contains call to operator.conditional
+        case _ => fail("AST Should contain one if statement")
+      }
+    }
+
+    "Contain a call to fmtString in the true branch of the <operator>.conditional" in {
+      inside(cpg.call.name(Operators.conditional).argument.l) {
+        case (condition: Call) :: (trueBranch: Block) :: _ :: Nil =>
+          condition.code shouldBe "ENV['USE_REDIS'] == 'true'"
+          condition.methodFullName shouldBe Operators.equals
+
+          inside(trueBranch.astChildren.isCall.l) {
+            case fmtStringCall :: Nil =>
+              fmtStringCall.methodFullName shouldBe Operators.formatString
+              fmtStringCall.argument.l.size shouldBe 4
+              val List(fmtVal1: Call, fmtVal2: Call, fmtVal3: Call, fmtVal4: Call) =
+                fmtStringCall.argument.l: @unchecked
+
+              fmtVal1.methodFullName shouldBe Operators.formattedValue
+              fmtVal1.code shouldBe "redis:host:"
```

Comment:
Why is this `redis.host:` and not just `host:`? This looks to me as if the .erb file is interpreted beyond the scope of the ERB handling.

Reply:
The new line gets removed in the `code`, so this is equivalent to
```
redis:
  host:
```
If you want I can try and change it so the newlines aren't removed and kept in the `code` property instead